packages/@azure/servicebus/data-plane/test/sessionsTests.spec.ts

```diff
@@ -348,10 +348,10 @@ describe("Session State", function(): void {
     should.equal(msgs[0].messageId, testMessage.messageId, "MessageId is different than expected");
     should.equal(msgs[0].sessionId, testMessage.sessionId, "SessionId is different than expected");
 
-    let testState = await receiver.getState();
+    let testState = await receiverClient.getSessionState(receiver.sessionId);
```

Comment:
Can you check if `getSessionState` and `setSessionState` work as expected when the session receiver is closed?
If not, then we should not move them out of the session receiver.

Reply:
Yes, I wanted to test this.
Will check and update.

Reply:
When the receiver is closed,
- `getSessionState()` and `setSessionState()` **failed** with
   >  SessionLockLostError: The session lock has expired on the MessageSession. Accept a new MessageSession. TrackingId:0e70ca15-e3a6-4055-9613-59d38267b441_B18, SystemTracker:NoSystemTracker, Timestamp:2019-02-26T06:24:56
    - This error came from the `messageCallback` of `sendRequestPromise` in `requestResponseLink.ts`(amqp-common) 
( as a result of the `sendRequest` made inside `managementClient.ts` ).

- `peekSession()` and `peekSessionBySequenceNumber()` still **succeeded**.
----------------------------------------------------
If the whole point of this issue is to reduce confusion to the user, I'd recommend not moving `getSessionState()` and `setSessionState()` as they won't work when the receiver is closed.

That means
- `getSessionState()` and `setSessionState()` will remain in the `receiver.ts` just like they used to be,  - `peekSession()` and `peekSessionBySequenceNumber()` will be moved to queueClient/subscriptionClient.

Reply:
Agreed

Reply:
But then, add test cases for when we peek with invalid session ids

Reply:
Noticed something strange.
I'm able to peek messages using `peekSessionBySequenceNumber` and `peekSession` for Unpartitioned entities even if invalid `sessionId` is passed.
For partitioned entities, null array is being returned if invalid `sessionId` is passed.

Reply:
> I'm able to peek messages using `peekSessionBySequenceNumber` and `peekSession` for Unpartitioned entities even if invalid `sessionId` is passed.

That is super weird. That could be a bug in Service Bus itself.
With this information, it looks like we dont have much to gain by moving peek out of the session receiver. Thoughts?

Reply:
The bug still remains no matter we move or not.

--------------------------------------------
(If) the purpose of moving peek out of session receiver is to reduce confusion on whether to call it by client or from the receiver of that client, i.e., 
```typescript
client.peekSession();
```
vs
```typescript
receiver = await receiverClient.getSessionReceiver();
receiver.peekSession();
```
it makes sense to call it from the client(moving out of session receiver), just like the usual peek.
[ User doesn't have to create a receiver if he just wants to peek ]

Reply:
> The bug still remains no matter we move or not

If the peek remains inside the session receiver, then there is no chance of user ever using peek with invalid session id

Reply:
Oh yes, got it.
But, shouldn't we be raising an issue to fix this problem even if it is hidden ?

Reply:
Yes, we should. What I am saying is we need to make the changes in this PR and expose the problem to our users